app/process/watcher/prometheus/watcher.go

```diff
@@ -101,98 +113,116 @@ func (pw Watcher) beginWatching() {
-		asset)
-	pw.initAndRegAssetMetric(asset, pw.supplyAssetsMetrics, name, help)
-}
-
-func (pw Watcher) registerBridgeAccAssetsMetrics(asset string) {
-	name := fmt.Sprintf("%s%s",
-		constants.BridgeAccAssetMetricsNamePrefix,
-		tokenIDtoMetricName(asset))
-	help := fmt.Sprintf("%s%s",
-		constants.BridgeAccAssetMetricsNameHelp,
-		asset)
-	pw.initAndRegAssetMetric(asset, pw.bridgeAccAssetsMetrics, name, help)
-}
-
-func (pw Watcher) registerEvmAssetSupplyMetric(asset string, name string) {
-	assetMetricName := fmt.Sprintf("%s%s",
-		constants.SupplyAssetMetricNamePrefix,
-		tokenIDtoMetricName(asset))
-	help := fmt.Sprintf("%s%s", constants.SupplyAssetMetricsHelpPrefix, name)
-	pw.initAndRegAssetMetric(asset, pw.supplyAssetsMetrics, assetMetricName, help)
-}
-
-func (pw Watcher) registerEvmAssetBalanceMetric(asset string, name string, address string) {
-	metricName := fmt.Sprintf("%s%s",
-		constants.BalanceAssetMetricNamePrefix,
-		tokenIDtoMetricName(asset))
-	help := fmt.Sprintf("%s%s%s%s",
-		constants.BalanceAssetMetricHelpPrefix,
-		name,
-		constants.AssetMetricHelpSuffix,
-		address)
-	pw.initAndRegAssetMetric(asset, pw.balanceAssetsMetrics, metricName, help)
+func (pw Watcher) regAssetMetric(
+	nativeNetworkId int64,
+	wrappedNetworkId int64,
+	assetAddress string,
+	metricNameCnt string,
+	metricHelpCnt string,
+) {
+	if assetAddress != constants.Hbar { // skip HBAR
+		assetName, assetSymbol := pw.getAssetRegData(wrappedNetworkId, assetAddress)
+		metricName, metricHelp := getMetricData(
+			nativeNetworkId,
+			wrappedNetworkId,
+			assetAddress,
+			assetName,
+			metricNameCnt,
+			metricHelpCnt,
+		)
+		if pw.assetsMetrics[wrappedNetworkId] == nil {
+			pw.assetsMetrics[wrappedNetworkId] = make(map[string]string)
+		}
+		pw.assetsMetrics[wrappedNetworkId][assetAddress] = metricName
+		pw.prometheusService.CreateGaugeIfNotExists(prometheus.GaugeOpts{
+			Name: metricName,
+			Help: metricHelp,
+			ConstLabels: prometheus.Labels{
+				constants.AssetMetricLabelKey: assetSymbol,
+			},
+		})
+	}
 }
 
-func tokenIDtoMetricName(id string) string {
-	replace := metrics.PrepareIdForPrometheus(id)
-	result := fmt.Sprintf("%s%s", constants.AssetMetricsNamePrefix, replace)
-	return result
+func (pw Watcher) getAssetRegData(networkId int64, assetAddress string) (name string, symbol string) {
+	if networkId == constants.HederaNetworkId { // Hedera
+		asset, e := pw.mirrorNode.GetToken(assetAddress)
+		if e != nil {
+			panic(e)
+		}
+		name = asset.Name
+		symbol = asset.Symbol
+	} else { // EVM
+		evm := pw.EVMClients[networkId].GetClient()
+		evmAssetInstance, e := wtoken.NewWtoken(common.HexToAddress(assetAddress), evm)
+		if e != nil {
+			panic(e)
+		}
+		resName, e := evmAssetInstance.Name(&bind.CallOpts{})
+		if e != nil {
+			panic(e)
+		}
+		name = resName
+		resSymbol, e := evmAssetInstance.Symbol(&bind.CallOpts{})
+		if e != nil {
+			panic(e)
+		}
+		symbol = resSymbol
+	}
+	return name, symbol
```

Comment:
I don't think it's a good idea to panic anywhere. If the provider timeouts it will crash the whole application